Zend/zend_closures.c

```diff
@@ -344,11 +344,7 @@ ZEND_METHOD(Closure, fromCallable)
 		RETURN_ZVAL(callable, 1, 0);
 	}
 
-	/* create closure as if it were called from parent scope */
-	EG(current_execute_data) = EX(prev_execute_data);
```

Comment:
May be just set the first parent user scope at this point?

Reply:
This code was already working correctly -- it worked around the bug in the is_callable implementation by manually going one frame up. However, the same issue exists for all other places that do callability checks in methods (including zpp).